### ISSUE TITLE
Feature Request: Add support for hyperlinks with `-R`

### ISSUE THREAD

**Issue by tomprince** [gwsw/less#54]
There is a standard for displaying hyperlinks in terminals [here](https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda). It would be nice if less passed through those escapes when `-R` is passed, like it does colors.

(Originally reported as 308 on the [bug list](http://www.greenwoodsoftware.com/less/bugs.html))

There is a patch [here](https://bug779734.bugzilla-attachments.gnome.org/attachment.cgi?id=349890) (from [here](https://bugzilla.gnome.org/show_bug.cgi?id=779734)).

**Comment by poettering** [comment]
Note that bigger projects including gcc and systemd nowadays generate hyperlinks like that in their output. It would be excellent if less would handle this correctly!

(I am systemd maintainer)

**Comment by dandavison** [comment]
In case it helps the process of getting this feature into master, I have applied the patch (with minor modifications to fix the conflicts) on master, confirmed that hyperlinks are working, and opened a PR: https://github.com/gwsw/less/pull/75.

(My motivation for this is that I'm the author of an application that relies on `less` for paging and would like to use hyperlinks: https://github.com/dandavison/delta/pull/261.)

**Comment by torarnv** [comment]
If adding to what `-R` does is seen as too risky for backwards compatibility, perhaps a new environment variable similar to `LESSANSIENDCHARS`and `LESSANSIMIDCHARS` could be added that controls which control codes to pass through, defaulting to the existing  ANSI color escape sequences?

**Comment by gwsw** [comment]
This is implemented in 3bb781ea44461dea611c4b1c5b414f97fc396cbe. My tests with self-constructed files seem to work fine, but could someone who regularly uses files containing such hyperlinks confirm that the behavior is as expected?

**Comment by dandavison** [comment]
Thanks very much for working on this @gwsw. I have built 3bb781ea44461dea611c4b1c5b414f97fc396cbe locally and can confirm that it is working as expected for me.

(The way I tested was a bit idiosyncratic, but for the record I used [delta](https://github.com/dandavison/delta)  with the setting `hyperlinks = true` and issued `git show`. With the version of less that I had installed already, the hyperlink was displayed correctly only if I used the `-r` flag, but not with `-R`. But with the version of less built from  3bb781ea44461dea611c4b1c5b414f97fc396cbe the hyperlink was displayed correctly when I used `less -R` and also `less -r`. This was on MacOS under iTerm2.).
